crates/file-serve/src/lib.rs

```diff
@@ -100,17 +105,45 @@ impl Server {
         self.addr.as_str()
     }
 
+    /// Whether the server is running
+    pub fn is_running(&self) -> bool {
+        self.running.load(Ordering::Acquire)
+    }
+
     /// Start the webserver
-    pub fn serve(&self) -> Result<std::convert::Infallible, Error> {
+    pub fn serve(&self) -> Result<(), Error> {
```

Comment:
Could you add a comment in the PR/commit explaining the use case you have for this?

Generally, I even prefer the use cases to be discussed in an issue before we get to a PR

Reply:
> Assume that someone was developing an App

I can assume that from the code.  I was hoping for something more concrete but oh well